packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -40,56 +46,50 @@ const convertToUiChatMessage = (
   isLargeGroup: boolean
 ): ChatMessage => {
   return {
-    type: 'chat',
-    payload: {
-      createdOn: message.createdOn,
-      content: message.content?.message,
-      type: sanitizedMessageContentType(message.type),
-      status: !isLargeGroup && message.status === 'delivered' && isSeen ? 'seen' : message.status,
-      senderDisplayName: message.senderDisplayName,
-      senderId: message.sender !== undefined ? toFlatCommunicationIdentifier(message.sender) : userId,
-      messageId: message.id,
-      clientMessageId: message.clientMessageId,
-      editedOn: message.editedOn,
-      deletedOn: message.deletedOn
-    }
+    messageType: 'chat',
+    createdOn: message.createdOn,
+    content: message.content?.message,
+    contentType: sanitizedMessageContentType(message.type),
+    status: !isLargeGroup && message.status === 'delivered' && isSeen ? 'seen' : message.status,
+    senderDisplayName: message.senderDisplayName,
+    senderId: message.sender !== undefined ? toFlatCommunicationIdentifier(message.sender) : userId,
+    messageId: message.id,
+    clientMessageId: message.clientMessageId,
+    editedOn: message.editedOn,
+    deletedOn: message.deletedOn
   };
 };
 
-const convertToUiSystemMessage = (message: ChatMessageWithStatus): Message<'system'> => {
+const convertToUiSystemMessage = (message: ChatMessageWithStatus): SystemMessage => {
   const systemMessageType = message.type;
   if (systemMessageType === 'participantAdded' || systemMessageType === 'participantRemoved') {
     return {
-      type: 'system',
-      payload: {
-        createdOn: message.createdOn,
-        participants:
-          message.content?.participants
-            // TODO: In our moderator logic, we use undefined name as our displayName for moderator, which should be filtered out
-            // Once we have a better solution to identify the moderator, remove this line
-            ?.filter((participant) => participant.displayName && participant.displayName !== '')
-            .map(
-              (participant): CommunicationParticipant => ({
-                userId: toFlatCommunicationIdentifier(participant.id),
-                displayName: participant.displayName
-              })
-            ) ?? [],
-        type: systemMessageType,
-        messageId: message.id,
-        iconName: systemMessageType === 'participantAdded' ? 'PeopleAdd' : 'PeopleBlock'
-      }
+      messageType: 'system',
```

Comment:
any reason for moving away from message.type to message.messageType?
`message` prefix feels redundant.

Reply:
To disambiguate from these other fields: `systemMessageType` (which further discriminates `SystemMessage` variants) and `contentType` (a field in `ChatMessage`).

Reply:
Approved :)